app/controllers/concerns/translatable.rb

```diff
@@ -7,18 +7,28 @@ module Translatable
 
   private
 
-    def translation_params(params)
-      resource_model.globalize_attribute_names.select { |k, v| params.include?(k.to_sym) && params[k].present? }
+    def translation_params(resource_model)
+      enabled_translations.flat_map do |loc|
+        resource_model.translated_attribute_names.map do |attr_name|
+          resource_model.localized_attr_name_for(attr_name, loc)
+        end
+      end
     end
 
     def delete_translations
```

Comment:
Good point, I hadn't thought of that. I'll think about how we can avoid it.

Reply:
How about

```ruby
  included do
    after_action :delete_translations, only: [:update], if: :update_was_successful?
  end

  private

    def update_was_successful?
      resource.valid?
    end
```

I think that should cover most "unsuccesful update" cases. Controllers can override the method to set different criteria. 

I would also change the name to `delete_unused_translations`.

Reply:
I think that condition would be OK in most cases. The site customization part is different, though, since it doesn't use the typical `if resource.update(params)` structure. Would the code still work as expected in that case?

+1 for changing the name to `delete_unused_translations` :+1:.